Source/APSlider.cpp

```diff
@@ -0,0 +1,143 @@
+/*
+  ==============================================================================
+
+    CustomSlider.cpp
+    Created: 20 Jan 2021 4:38:04pm
+    Author:  Johnathan Handy
+
+  ==============================================================================
+*/
+
+#include "APSlider.h"
+
+using namespace juce;
+
+APSlider::APSlider(Ap_dynamicsAudioProcessor& p, SliderType sliderType) : audioProcessor (p), sliderType_ (sliderType)
+{
+    ratioSliderBar_.start();
+    addAndMakeVisible (ratioSliderBar_);
+    threshSliderBar_.start();
+    addAndMakeVisible (threshSliderBar_);
+    addAndMakeVisible (slider);
+
+    startTimerHz(30);
+}
+
+APSlider::~APSlider()
+{
+    ratioSliderBar_.stop();
+    threshSliderBar_.stop();
+    stopTimer();
+}
+
+void APSlider::resized()
+{
+    auto offset = 72.0f;
+    switch (sliderType_) {
+        case Normal:
+            ratioBounds_ = Rectangle<int> (1, 10,
+                                        getWidth() - offset, ap_slider_height - 20);
+            ratioSliderBar_.setBounds(ratioBounds_);
+            slider.setBounds(getLocalBounds());
+            break;
+        case Invert:
+//            ratioBounds_ = Rectangle<int> (1, 10,
+//                                        getWidth() - offset, getHeight() - 20);
+            threshBounds_ = Rectangle<int> (1, 10,
+                                        getWidth() - offset, ap_slider_height - 20);
+//            ratioSliderBar_.setBounds(ratioBounds_);
+            threshSliderBar_.setBounds(threshBounds_);
+            slider.setBounds(getLocalBounds());
+            break;
+        default:
+            break;
```

Comment:
If there are no other slider types, then put `jassertfalse` into the default case.